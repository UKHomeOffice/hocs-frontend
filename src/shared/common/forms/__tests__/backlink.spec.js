import React from 'react';
import Backlink from '../backlink';
import { MemoryRouter } from 'react-router-dom';

describe('Form backlink component', () => {

    const mockDispatch = jest.fn();

    beforeEach(() => {
        mockDispatch.mockReset();
    });

    it('should render with default props', () => {
        const outer = shallow(<Backlink action="/SOME/URL" />);
        const Children = outer.props().children;
        const wrapper = mount(
            <MemoryRouter>
                <Children dispatch={mockDispatch} />
            </MemoryRouter>
        );
        expect(wrapper).toBeDefined();
        expect(wrapper.find('BackLink')).toMatchSnapshot();
    });

    it('should render disabled when isDisabled is passed', () => {
        const outer = shallow(<Backlink disabled={true} action="/SOME/URL" />);
        const Children = outer.props().children;
        const wrapper = mount(
            <MemoryRouter>
                <Children dispatch={mockDispatch} />
            </MemoryRouter>
        );
        expect(wrapper).toBeDefined();
        expect(wrapper.find('BackLink')).toMatchSnapshot();
        expect(wrapper.find('Link').props().disabled).toEqual(true);
    });

    it('should render with correct label when passed', () => {
        const outer = shallow(<Backlink label={'My Button'} action="/SOME/URL" />);
        const Children = outer.props().children;
        const wrapper = mount(
            <MemoryRouter>
                <Children dispatch={mockDispatch} />
            </MemoryRouter>
        );
        expect(wrapper).toBeDefined();
        expect(wrapper.find('BackLink')).toMatchSnapshot();
        expect(wrapper.find('Link').props().children).toEqual('My Button');
    });

    it('should render with additional styles when className is passed', () => {
        const outer = shallow(<Backlink className={'test-class'} action="/SOME/URL" />);
        const Children = outer.props().children;
        const wrapper = mount(
            <MemoryRouter>
                <Children dispatch={mockDispatch} />
            </MemoryRouter>
        );
        expect(wrapper).toBeDefined();
        expect(wrapper.find('BackLink')).toMatchSnapshot();
        expect(wrapper.find('Link').props().className).toEqual('govuk-back-link test-class');
    });

    it('should dispatch action when clicked', () => {
        const outer = shallow(<Backlink action="/SOME/URL" />);
        const Children = outer.props().children;
        const wrapper = mount(
            <MemoryRouter>
                <Children dispatch={mockDispatch} />
            </MemoryRouter>
        );
        expect(wrapper).toBeDefined();
        wrapper.find('Link').simulate('click');
        expect(mockDispatch).toHaveBeenCalledTimes(1);
    });

});