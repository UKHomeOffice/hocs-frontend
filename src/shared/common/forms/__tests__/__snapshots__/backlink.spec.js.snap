// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form backlink component should render disabled when isDisabled is passed 1`] = `
<BackLink
  action="/SOME/URL"
  disabled={true}
  dispatch={[MockFunction]}
  label="Back"
>
  <p>
    <Link
      className="govuk-back-link"
      disabled={true}
      onClick={[Function]}
      replace={false}
      to="/SOME/URL"
    >
      <a
        className="govuk-back-link"
        disabled={true}
        href="/SOME/URL"
        onClick={[Function]}
      >
        Back
      </a>
    </Link>
  </p>
</BackLink>
`;

exports[`Form backlink component should render with additional styles when className is passed 1`] = `
<BackLink
  action="/SOME/URL"
  className="test-class"
  disabled={false}
  dispatch={[MockFunction]}
  label="Back"
>
  <p>
    <Link
      className="govuk-back-link test-class"
      disabled={false}
      onClick={[Function]}
      replace={false}
      to="/SOME/URL"
    >
      <a
        className="govuk-back-link test-class"
        disabled={false}
        href="/SOME/URL"
        onClick={[Function]}
      >
        Back
      </a>
    </Link>
  </p>
</BackLink>
`;

exports[`Form backlink component should render with correct label when passed 1`] = `
<BackLink
  action="/SOME/URL"
  disabled={false}
  dispatch={[MockFunction]}
  label="My Button"
>
  <p>
    <Link
      className="govuk-back-link"
      disabled={false}
      onClick={[Function]}
      replace={false}
      to="/SOME/URL"
    >
      <a
        className="govuk-back-link"
        disabled={false}
        href="/SOME/URL"
        onClick={[Function]}
      >
        My Button
      </a>
    </Link>
  </p>
</BackLink>
`;

exports[`Form backlink component should render with default props 1`] = `
<BackLink
  action="/SOME/URL"
  disabled={false}
  dispatch={[MockFunction]}
  label="Back"
>
  <p>
    <Link
      className="govuk-back-link"
      disabled={false}
      onClick={[Function]}
      replace={false}
      to="/SOME/URL"
    >
      <a
        className="govuk-back-link"
        disabled={false}
        href="/SOME/URL"
        onClick={[Function]}
      >
        Back
      </a>
    </Link>
  </p>
</BackLink>
`;
